llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1439,19 +1440,19 @@ void VPlanTransforms::truncateToMinimalBitwidths(
 }
 
 void VPlanTransforms::optimize(VPlan &Plan) {
-  removeRedundantCanonicalIVs(Plan);
-  removeRedundantInductionCasts(Plan);
+  runPass(removeRedundantCanonicalIVs, Plan);
+  runPass(removeRedundantInductionCasts, Plan);
 
   simplifyRecipes(Plan, Plan.getCanonicalIV()->getScalarType());
```

Comment:
How come we don't use runPass for simplfiyRecipes?

Reply:
Updated to also run `simplifyRecipes`; needed passing the canonical IV type by reference